src/main/java/org/xwiki/git/GitManagerNew.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.git;
+
+import org.eclipse.jgit.lib.Repository;
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Provides services to access a Private Git repository.
+ *
+ * @version $Id$
+ * @since 12.3RC1
+ */
+@Unstable
+@Role
+public interface GitManagerNew extends GitManager
```

Comment:
bad name....

Reply:
Should not be a new interface

Reply:
If I create the method inside GitManager, I get error on revapi-check. I am probably missing the perfect way to do this, can you provide such a way?

Reply:
Revapi is correct. You must not break backward compatibility! :) Please research it a bit, it should be easy to find the answer (if I tell you then I'm doing all the job and it won't prove your ability to research a topic and think about finding a solution). Thanks!

Reply:
Got it, thanks. I will keep trying until you're satisfied :)